src/app/frontend/deploy/deployfromfile.html

```diff
@@ -0,0 +1,34 @@
+<!--Copyright 2015 Google Inc.All Rights Reserved.
+
+     Licensed under the Apache License,
+    Version 2.0(the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+See the License for the specific language governing permissions and
+limitations under the License.
+-->
+
+<div layout="row" layout-align="space-between start">
+  <div flex="80">
+    <md-input-container class="md-block">
+      <label>YAML or JSON file</label>
+      <input ng-model="ctrl.file.name" ng-disabled="true"/>
```

Comment:
In this case two different elements will be interacting each other. Visible input field and hidden input field. There are 3 solutions. 
1- I can insert a custom attribute in the visible input field and query it from the kdFileReaderDirective's link function to register focus event. 
2- I can implement different directive for the visible input field which registers the focus event in its link function by using the id of the hidden input field. 
3- I can implement a new directive like kd-upload which contains all the components or elements (Input fields and button) required for the upload functionality. Those input fields interact each other via the scope of this directive. 
I think solution 3 is better. What do you think?